cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/io/cdap/cdap/etl/spark/batch/RDDCollection.java

```diff
@@ -123,6 +126,54 @@ public RDDCollection(JavaSparkExecutionContext sec, JavaSparkContext jsc, SQLCon
       }
       seenRequired = seenRequired || toJoin.isRequired();
 
+      // UUID for salt column name to avoid name collisions
+      String saltColumn = UUID.randomUUID().toString();
+      if (joinRequest.isDistributionEnabled() && JoinDistribution.getSupportedJoinTypes().contains(joinType)) {
+
+        final String saltUDF = "salt";
+        final String prepareExplodeUDF = "prepare_explode";
+
+        //Array of [0,distributionFactor) to be used in the PrepareExplode step
+        Integer[] numbers =
+          IntStream.range(0, joinRequest.getDistribution().getDistributionFactor()).boxed().toArray(Integer[]::new);
+
+        // Register UserDefinedFunctions for use later
+        sqlContext.udf().register(prepareExplodeUDF, prepareExplodeUDF(numbers),
+                                  DataTypes.createArrayType(DataTypes.IntegerType));
+        sqlContext.udf().register(saltUDF, saltUDF(joinRequest.getDistribution().getDistributionFactor()),
```

Comment:
could this be done without a UDF? maybe using a combination of rand, multiplication, and floor?